W-22405361-ETS-change (#307)
mulesoft/docs-mule-sdk: modules/ROOT/pages/publishing-connectors-exchange.adoc

= Publishing Connectors to Exchange
:page-id: publishing-connectors-exchange

Starting August 1, 2026, publishing connectors and Mule plugins to Exchange requires Java compatibility metadata in `mule-artifact.json` when publishing new versions that change supported Java compatibility.

Exchange reads Java compatibility from `mule-artifact.json` during publication.

[NOTE]

> Update your connector configuration before publishing a new connector version. This change doesn't affect existing connectors that are already published to Exchange.

Reviewer: ```suggestion
Update your connector configuration before publishing a new connector version. This change doesn't affect connectors that are already published to Exchange.
```

== Update Your Connector

> Use the path that matches your connector type:

Reviewer: ```suggestion
Use the requirements that match your connector type:
```

* Java connectors: Include the `@JavaVersionSupport` annotation when changing supported Java compatibility.
* XML connectors: Update to `mule-extensions-maven-plugin` so Java compatibility metadata is generated at build time.

> Choose the approach that matches your development setup:

Reviewer: ```suggestion
For XML connectors, choose the approach that matches your development setup:
```

Reviewer: i understood that these steps were for XML connectors? If not, dismiss this.

* If you use the parent POM, update to a parent version that includes `mule-extensions-maven-plugin` 1.9.0 or later.
+
[source,xml]
----
<parent>
    <groupId>org.mule.connectors</groupId>
    <artifactId>mule-modules-parent</artifactId>
    <version>1.9.0</version> <!-- or later -->
</parent>
----

* If you use Mule SDK, upgrade to a version that includes `mule-extensions-maven-plugin` 1.9.0 or later.

> * If you manually configure plugins, update your POM file to reference `mule-extensions-maven-plugin` 1.9.0 or later.

Reviewer: ```suggestion
* If you configure Maven plugins manually, update your POM file to reference `mule-extensions-maven-plugin` 1.9.0 or later.
```
